rails/app/models/speaker.rb

```diff
@@ -12,13 +29,24 @@ def picture_url
 
   def self.import_csv(filename)
     CSV.parse(filename, headers: true) do |row|
-      speaker = Speaker.where(name: row[0], community: row[2]).first_or_create
+      speaker = Speaker.find_or_create_by(
+        name: row[0], 
+        birth_year: row[1].nil? ? nil : Date.strptime(row[1], "%Y"), 
+        birthplace: get_birthplace(row[2])
+      )
       if row[3] && File.exist?(Rails.root.join('media', row[3]))
         file = File.open(Rails.root.join('media',row[3]))
         speaker.media.attach(io: file, filename: row[3])
+      
         speaker.save
       end
     end
   end
 
+  def self.get_birthplace(name)
+    if name.nil? || name.downcase == 'unknown'
+      return nil
+    end
+    return Place.find_or_create_by(name: name)
+  end
```

Comment:
NIT but you could change this to just `Place.find_or_create_by(name: name)` since Ruby methods automatically return the value of the last expression

Reply:
ah yes!!!  too much time in javascript land :P